uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -1265,6 +1272,14 @@ public synchronized void deletePipelineInMirrorMaker(String pipeline) {
     _lock.lock();
     try {
       LOGGER.info("Trying to delete pipeline: {}", pipeline);
+      PriorityQueue<InstanceTopicPartitionHolder> itphSet = _pipelineToInstanceMap.get(pipeline);
+      for (InstanceTopicPartitionHolder itph : itphSet) {
+        if (itph.getTotalNumPartitions() != 0) {
+          throw new UnsupportedOperationException("Delete non-empty pipeline is not allowed, serving number of partitions: "
```

Comment:
Can you explain why disallow deleting non-empty pipeline? When can this happen, and what action to take when it happens?

Reply:
this action might be trigger manually for pipeline cleanup. we might accidentally delete non-empty pipeline without this check, if so, topic in that pipeline will not been replicated.

Reply:
got it. thanks